core/src/main/java/org/eclipse/hono/config/AuthenticatingClientConfigProperties.java

```diff
@@ -29,14 +30,15 @@
 
     public static final String SERVER_ROLE_UNKNOWN = "unknown";
 
-    private String credentialsPath;
+    private String credentialsPath = "";
     private String host = "localhost";
     private boolean hostnameVerificationRequired = true;
-    private char[] password;
-    private int port;
+    private String password = "";
+    private int port = Constants.PORT_AMQPS;
```

Comment:
what's the reason for having to use an empty string as the initial value?

Reply:
@sophokles73 At the moment with way we use Quarkus, it is complaining when properties don't have default values. So this is a workaround. I haven't yet raised this with the Quarkus devs, but I can do that and we can revert it in the future (I'm not too fond of it myself either).

Reply:
yes, please do so. Honestly, I hate the guts out of it ;-)

Reply:
Setting them explicitly to null also does the trick and it looks a bit nicer. So let's go with that for the moment and push for the fix.